packages/plugins/plugin-script/src/components/NotebookStack/NotebookStack.tsx

```diff
@@ -17,74 +32,171 @@ import { type TypescriptEditorProps } from '../TypescriptEditor';
+        <StackItem.DragHandle>
+          <div className='flex justify-center p-2'>
+            <Icon icon='ph--dots-six-vertical--regular' size={4} />
+          </div>
+        </StackItem.DragHandle>
+        <DropdownMenu.Root>
+          <DropdownMenu.Trigger asChild>
+            <div className='flex justify-center p-2'>
+              <Icon icon='ph--dots-three--regular' size={4} />
+            </div>
+          </DropdownMenu.Trigger>
+          <DropdownMenu.Portal>
+            <DropdownMenu.Content>
+              <DropdownMenu.Viewport>
+                <DropdownMenu.Item onClick={() => onCellInsert?.('script', cell.id)}>
+                  {t('notebook cell insert script label')}
+                </DropdownMenu.Item>
+                <DropdownMenu.Item onClick={() => onCellInsert?.('prompt', cell.id)}>
+                  {t('notebook cell insert prompt label')}
+                </DropdownMenu.Item>
+                <DropdownMenu.Item onClick={() => onCellInsert?.('query', cell.id)}>
+                  {t('notebook cell insert query label')}
+                </DropdownMenu.Item>
+                <DropdownMenu.Item onClick={() => onCellInsert?.('markdown', cell.id)}>
+                  {t('notebook cell insert markdown label')}
+                </DropdownMenu.Item>
+                <DropdownMenu.Item onClick={() => onCellDelete?.(cell.id)}>
+                  {t('notebook cell delete label')}
+                </DropdownMenu.Item>
+              </DropdownMenu.Viewport>
+              <DropdownMenu.Arrow />
+            </DropdownMenu.Content>
+          </DropdownMenu.Portal>
+        </DropdownMenu.Root>
       </StackItem.Heading>
 
-      <StackItem.Content>
+      {/* TODO(burdon): Fix drag preview. */}
+      <StackItem.DragPreview>
+        {({ item: cell }) => (
+          <StackItem.Content classNames='overflow-visible bg-groupSurface border border-subduedSeparator'>
+            <NotebookCell cell={cell} space={space} env={env} />
+          </StackItem.Content>
+        )}
+      </StackItem.DragPreview>
+
+      <StackItem.Content classNames='overflow-visible'>
+        <NotebookCell cell={cell} space={space} env={env} onCellRun={onCellRun} />
+        {(value != null || view != null) && (
+          <div className='flex is-full bg-groupSurface border-t border-subduedSeparator text-description font-mono'>
+            {value != null && (
+              <div className={mx(editorStyles)}>
+                {name && (
+                  <>
+                    <span className='text-successText'>{name}</span>
+                    <span className='text-description'>&nbsp;=&nbsp;</span>
+                  </>
+                )}
+                <span>{value}</span>
+              </div>
+            )}
+            {view && (
+              <div className='flex is-full overflow-hidden'>
+                <Surface role='section' limit={1} data={{ subject: view }} />
+              </div>
```

Comment:
Surfaces fulfilling role section expect to be direct descendants of a StackItem.Root in a vertical stack – putting them in an arbitrary div with flex is bound to break. What's the intention here?

Reply:
no